src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -114,6 +114,67 @@ Kokkos::View<size_t *, Kokkos::Device<Kokkos::Cuda, MemorySpace>> sortObjects(
   return permute;
 }
 #endif
+
+// Helper functions and structs for applyPermutations
+namespace PermuteHelper
+{
+template <class DstViewType, class SrcViewType, int Rank = DstViewType::Rank>
+struct CopyOp;
```

Comment:
I don't think you need to specialization. I think something like this should work:
```c++
KOKKOS_INLINE_FUNCTION
static void copy(DstViewType const &dst, size_t i_dst, SrcViewType const &src,
                 size_t i_src)
{
  for (unsigned int j = 0; j < dst.extent(1); ++j)
    for (unsigned int k = 0; k < dst.extent(2); ++k)
      for (unsigned int m = 0; m < dst.extent(3); ++m)     
         dst.access(i_dst, j, k, m) = src.access(i_src, j, k, m);
}
```
Note that I am not advocating this solution. I just wanted to point out that this was possible.